automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/apps/AppCreationSpec.scala

```diff
@@ -192,20 +195,26 @@ class AppCreationSpec extends GPAllocFixtureSpec with LeonardoTestUtils with GPA
           listApps = LeonardoApiClient.listApps(googleProject, true)
           implicit0(doneCheckable: DoneCheckable[List[ListAppResponse]]) = appDeleted(appName)
           monitorDeleteResult <- streamFUntilDone(listApps, 120, 10 seconds).compile.lastOrError
+          // TODO remove first case in below if statement when Galaxy deletion is reliable
           _ <- if (!doneCheckable.isDone(monitorDeleteResult)) {
             loggerIO.warn(
               s"AppCreationSpec: app ${googleProject.value}/${appName.value} did not finish deleting after 20 minutes. Result: $monitorDeleteResult"
             )
           } else {
-            loggerIO.info(
-              s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorDeleteResult"
-            )
+            // verify disk is also deleted
+            for {
+              _ <- loggerIO.info(
+                s"AppCreationSpec: app ${googleProject.value}/${appName.value} delete result: $monitorDeleteResult"
+              )
+              getDiskResp <- streamUntilDoneOrTimeout(
```

Comment:
I made this retry the `getDisk` call.. it looks like monitoring `DiskStatus.Deleting -> DiskStatus.Deleted` happens asynchronously to moving the app to `Deleted`. So there was a race condition here.

Reply:
I don't think the deletion happens async, no? see [here](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala#L995)

Reply:
think I'm going to remove the wait for now and monitor tests (also need to add back the assertion)

Reply:
Hm yeah originally I thought disk monitoring happened async but I guess not since `sync=true`.

Maybe this failure was test code related too.. if Galaxy deletion fails then the disk would get stuck in `Deleting`. In that case moving this check to the `else` block here should help too.

Reply:
oh yea....I'm definitely moving the check into the branch(thanks for spotting that), just don't think the wait is needed